pkg/agent/multicast/mcast_controller.go

```diff
@@ -473,6 +478,109 @@ func (c *Controller) addOrUpdateGroupEvent(e *mcastGroupEvent) {
 	}
 }
 
+func (c *Controller) memberChanged(e interface{}) {
+	podEvent := e.(types.PodUpdate)
+	namespace, name := podEvent.PodNamespace, podEvent.PodName
+	containerID := podEvent.ContainerID
+
+	klog.V(2).InfoS("Calling memberChanged", "IsAdd", podEvent.IsAdd, "namespace", namespace,
+		"name", name)
+	if podEvent.IsAdd {
+		iface, ok := c.ifaceStore.GetContainerInterface(containerID)
+		if ok {
+			g := &mcastGroupEvent{
+				group: types.McastAllHosts,
+				eType: groupJoin,
+				iface: iface,
+			}
+			c.groupEventCh <- g
+		}
+	} else {
+		interfaceName := util.GenerateContainerInterfaceName(name, namespace, containerID)
+		ifaceConfig := &interfacestore.InterfaceConfig{
+			InterfaceName: interfaceName,
+		}
+		g := &mcastGroupEvent{
+			group: types.McastAllHosts,
+			eType: groupLeave,
+			iface: ifaceConfig,
+		}
+		c.groupEventCh <- g
+		c.removeLocalInterface(ifaceConfig)
+	}
+}
+
+func (c *Controller) syncQueryGroup(event *mcastGroupEvent) {
+	if c.queryGroupId == 0 {
+		klog.V(2).InfoS("As c.queryGroupId is 0, query group will not be installed")
+		return
+	}
+	switch event.eType {
+	case groupJoin:
+		if event.iface != nil {
+			memberPorts := make([]uint32, 0, len(c.queryGroupMember)+1)
+			for _, port := range c.queryGroupMember {
+				memberPorts = append(memberPorts, port)
+			}
+			memberPorts = append(memberPorts, uint32(event.iface.OFPort))
+			err := c.updateQueryGroup(memberPorts)
+			if err != nil {
+				klog.ErrorS(err, "Failed to update query group for groupJoin")
```

Comment:
It is not good to ignore the error and then update the memory cacche, even if an error log is added. The result is the memory cache is inconsistent with datapath. So does the delete logic.

Reply:
Do you think `queryGroupMember` is really needed? Can we use the interfaceStore as the cache?

Reply:
OK, removed